Entity/Components/Stats/StatModifiers.gd

```diff
@@ -0,0 +1,32 @@
+class_name StatModifiers extends Resource
+
+var modifiers: Dictionary = {}
+
```

Comment:
So just to confirm my understanding of the design here:
- temporary = lasts until combat end
- permanent = buffs your character permanently
So I'm guessing the idea behind permanent buffs is you get them from stuff like artifacts that last for the entire run?

Reply:
Exactly. Not to say you can't lose the buff (for example if you lose the object for whatever reason), but as you can see in the reset function, I only reset the temporary modifiers.

Reply:
I see. In that case, there may need to be some changes whenever we implement #35, because the player will get destroyed at the end of combat and therefore the StatComponent will also be destroyed, meaning all the stat objects will be garbage collected. Maybe we can change it to where stats are decoupled from the entity, and there's a global stat holder that keeps stats for the duration of a single run. Maybe in `PlayerManager`

Reply:
Ah indeed, I did not realize that the player entity was destroyed at combat end. We would need to hold stats separate from the entity then. The `PlayerManager` indeed seems like the good place to do so. Isn't it the same problem for health though ?

Reply:
That's a good point. We'll probably have to store our health in `PlayerManager` too

Reply:
This should probably be done in a separate issue/PR before #35